docs/src/main/sphinx/security/user-mapping.rst

```diff
@@ -2,67 +2,123 @@
-Username and Password (file or LDAP)  ``http-server.authentication.password.user-mapping.pattern``
-Kerberos                              ``http-server.authentication.krb5.user-mapping.pattern``
+Password (file, LDAP, Salesforce)     ``http-server.authentication.password.user-mapping.pattern``
+OAuth2                                ``http-server.authentication.oauth2.user-mapping.pattern``
 Certificate                           ``http-server.authentication.certificate.user-mapping.pattern``
-Json Web Token                        ``http-server.authentication.jwt.user-mapping.pattern``
+JSON Web Token                        ``http-server.authentication.jwt.user-mapping.pattern``
+Kerberos                              ``http-server.authentication.krb5.user-mapping.pattern``
 Insecure                              ``http-server.authentication.insecure.user-mapping.pattern``
 ===================================== ===============================================================
 
+.. _pattern-file:
+
 File mapping rules
 ------------------
 
-The file mapping rules allow for more complex mappings from the authentication user.  These rules are
-loaded from a JSON file defined in a configuration property.  The mapping is based on the first matching
-rule, processed from top to bottom. If no rules match, authentication is denied.  Each rule is composed of the
+Use the **File mapping rules** method if your authentication provider’s way to
+express usernames cannot be reduced to a single rule, or if you want to exclude
+a set of users from accessing the cluster.
+
+The rules are loaded from a JSON file identified in a configuration property.
+The mapping is based on the first matching rule, processed from top to bottom.
+If no rules match, authentication is denied.  Each rule is composed of the
 following fields:
 
-* ``pattern`` (required): regex to match against authentication user.
-* ``user`` (optional): replacement string to substitute against pattern.
+* ``pattern`` (required): regex to match against the authentication method's
+  username.
+* ``user`` (optional): replacement string to substitute against *pattern*.
   The default value is ``$1``.
-* ``allow`` (optional): boolean indicating if the authentication should be allowed.
+* ``allow`` (optional): Boolean indicating whether authentication is to be
+  allowed for the current match.
 * ``case`` (optional): one of:
 
-  * ``keep`` - keep matched user name as is (default behavior)
-  * ``lower`` - lowercase matched user name, e.g., ``Admin`` or ``ADMIN`` will become ``admin``
-  * ``upper`` - uppercase matched user name, e.g., ``admin`` or ``Admin`` will become ``ADMIN``
+  * ``keep`` - keep the matched username as is (default behavior)
+  * ``lower`` - lowercase the matched username; thus both ``Admin`` and ``ADMIN`` become ``admin``
+  * ``upper`` - uppercase the matched username; thus both ``admin`` and ``Admin`` become ``ADMIN``
```

Comment:
Missing periods

Reply:
See above about Dr. Google and his Style Guide Minions.